plugins/circleci/src/pages/BuildsPage/lib/Builds/Builds.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in wr iting, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { FC } from 'react';
+
+import { BuildSummary } from 'circleci-api';
+
+import { CITable, CITableBuildInfo } from '../CITable';
+import { useSelector, useDispatch } from 'react-redux';
+import { iRootState, Dispatch } from 'state/store';
+import { useApi } from '@backstage/core';
+import { circleCIApiRef } from 'api';
+
+const makeReadableStatus = (status: string | undefined) => {
+  if (typeof status === 'undefined') return '';
+  return ({
+    retried: 'Retried',
+    canceled: 'Canceled',
+    infrastructure_fail: 'Infra fail',
+    timedout: 'Timedout',
+    not_run: 'Not run',
+    running: 'Running',
+    failed: 'Failed',
+    queued: 'Queued',
+    scheduled: 'Scheduled',
+    not_running: 'Not running',
+    no_tests: 'No tests',
+    fixed: 'Fixed',
+    success: 'Success',
+  } as Record<string, string>)[status];
```

Comment:
That's exactly the problem here, since the `status` has wider type of `string` and cannot be used as key type on this object (which is a discriminated union of string literals indeed)

Reply:
So then status should have a more narrow type? :grin:

Reply:
That's correct! Buuuut, this status comes originally comes from the `circleci-api` lib itself, where (guess what) they declared it as a `string` :) So we can make it typed in our code, but that defeats the purpose of delegating the API layer implementation to the separate lib

Reply:
Yeah figured it was from an API, adding types was more about avoiding typos in this project. I'm fine with not fixing it though. I would say cast to `{ [key in string]: string }` instead, we usually use that form.